lib/_stream_readable.js

```diff
@@ -152,6 +154,28 @@ Readable.prototype.isPaused = function() {
   return this._readableState.flowing === false;
 };
 
+Readable.prototype.destroy = function() {
+  var self = this;
+  if (this._readableState.destroyed) return;
+  this._readableState.destroyed = true;
+
+  // if this is a duplex stream mark the writable part as destroyed as well
+  if (this._writableState) {
+    this._writableState.destroyed = true;
+  }
```

Comment:
It's almost a shame we couldn't add this to the base `stream` class somehow since they are both doing pretty much the same thing.

Better still if there was a base `StreamState` that all states inherited from, although that would restrict people from calling `_destroy()` directly, although that might not be such a bad thing...